apm-agent-plugins/apm-redis-plugin/apm-lettuce-plugin/src/main/java/co/elastic/apm/agent/redis/lettuce/Lettuce5StartSpanInstrumentation.java

```diff
@@ -0,0 +1,94 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.redis.lettuce;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.redis.RedisSpanUtils;
+import com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap;
+import io.lettuce.core.protocol.RedisCommand;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.nameEndsWith;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+/**
+ * Starts a span in {@link io.lettuce.core.RedisChannelHandler#dispatch(RedisCommand)}
+ *
+ * The context will be propagated via the Netty instrumentation
+ */
+public class Lettuce5StartSpanInstrumentation extends ElasticApmInstrumentation {
```

Comment:
Can this be a subclass of a common super with the 3.4 abstract instrumentation? Seems it would only depend on `io.lettuce.core.protocol.RedisCommand` which is common. Then you can share the map and the instrumentation groups at least among all of them.

Reply:
Unfortunately not. In Lettuce 3 and 4, the class is called `com.lambdaworks.redis.protocol.RedisCommand`, in Lettuce 5 it's `io.lettuce.core.protocol.RedisCommand`

Reply:
Ahh, I looked at the imports of `Lettuce34Instrumentation`